include/deal.II/base/parameter_handler.h

```diff
@@ -1176,212 +1176,236 @@ namespace Patterns
+ *     void get_parameters (ParameterHandler &prm);
+ *   private:
+ *     std::string method;
+ *     int         max_iterations;
+ *   };
+ *
+ *
+ *
+ *   class Problem
+ *   {
+ *   private:
+ *     LinearEquation eq1, eq2;
+ *     std::string matrix1, matrix2;
+ *     std::string outfile;
+ *   public:
+ *     static void declare_parameters (ParameterHandler &prm);
+ *     void get_parameters (ParameterHandler &prm);
+ *
+ *     void do_something ();
+ *   };
+ *
+ *
+ *
+ *   void LinearEquation::declare_parameters (ParameterHandler &prm)
+ *   {
+ *     // declare parameters for the linear solver in a subsection
+ *     prm.enter_subsection ("Linear solver");
+ *     prm.declare_entry ("Solver",
+ *                        "CG",
+ *                        Patterns::Selection("CG|BiCGStab|GMRES"),
+ *                        "Name of a linear solver for the inner iteration");
+ *     prm.declare_entry ("Maximum number of iterations",
+ *                        "20",
+ *                        Patterns::Integer());
+ *     prm.leave_subsection ();
+ *   }
  *
- *     using namespace dealii;
- *
- *     class LinEq
- *     {
- *     public:
- *       static void declare_parameters (ParameterHandler &prm);
- *       void get_parameters (ParameterHandler &prm);
- *     private:
- *       std::string Method;
- *       int    MaxIterations;
- *     };
  *
  *
- *     class Problem
- *     {
- *     private:
- *       LinEq eq1, eq2;
- *       std::string Matrix1, Matrix2;
- *       std::string outfile;
- *     public:
- *       static void declare_parameters (ParameterHandler &prm);
- *       void get_parameters (ParameterHandler &prm);
- *     };
- *
+ *   void LinearEquation::get_parameters (ParameterHandler &prm)
+ *   {
+ *     prm.enter_subsection ("Linear solver");
+ *     method        = prm.get ("Solver");
+ *     max_iterations = prm.get_integer ("Maximum number of iterations");
+ *     prm.leave_subsection ();
+ *     std::cout << "  LinearEquation: method=" << method
+ *               << ", max_iterations=" << max_iterations
+ *               << std::endl;
+ *   }
+ *
+ *
+ *
+ *   void Problem::declare_parameters (ParameterHandler &prm)
+ *   {
+ *     // first some global parameter entries
+ *     prm.declare_entry ("Output file",
+ *                        "out",
+ *                        Patterns::Anything(),
+ *                        "Name of the output file, either relative to the present "
+ *                        "path or absolute");
+ *     prm.declare_entry ("Equation 1",
+ *                        "Laplace",
+ *                        Patterns::Anything(),
+ *                        "String identifying the equation we want to solve");
+ *     prm.declare_entry ("Equation 2",
+ *                        "Elasticity",
+ *                        Patterns::Anything());
+ *
+ *     // declare parameters for the first equation
+ *     prm.enter_subsection ("Equation 1 Settings");
+ *     prm.declare_entry ("Matrix type",
```

Comment:
We've more recently started to use a style where we enclose things that are in a subsection into braces, i.e.,
```
  prm.enter_subsection (...);
  {
    prm.declare_entry (...);
  }
  prm.leave_subsection;
```
This has the advantage that it also visually indicates what is logically happening. Would you mind updating the example this way?